db/controll.js

```diff
@@ -0,0 +1,28 @@
+
+var Search=require('../db/models/model');
+
+
+
+
+let fetchData=function(callback){
+  Search.find({},function(err,data){
+    if(err){
+      console.log(err)
+    }else{
+
+      callback(data)
+    }
+  }).sort({'id':1}).limit(10)
+}
+
+let findOne=function(query,callback){
+  Search.find(query,(err,data)=>{
+    if(err){
+      console.log(err);
+    }else{
+      callback(data);
+    }
+  }).sort({'id':1}).limit(10)

```

Comment:
`.limit(10)` is a bit confusing given this function is supposed to find one. A `.limit(1)` would also have made the api call test output return only the first result for the searched title. It also may be more appropriate to rename the `query` parameter to `title`.

Reply:
sorry for not clear first, my goal is when we get the search result from db, i also tell server to get all related catagory based on this search in db, so i am trying to get all related title or catagory in this first run, , yeah, i will clean up the code , my component is setting up the suggestion and ads for every search results

Reply:
👍